### PR TITLE
Fixed maxOptions losing focus

### PR DESCRIPTION
**Pull request by choisee**
Fixed losing focus when using maxOptions - #73

### CODE REVIEW

**Comment by choisee** [comment]
@yury-dymov Hi, Can you review this please?

File: src/AutoCompleteTextField.js [hunk 1]
```diff
@@ -297,7 +297,8 @@ class AutocompleteTextField extends React.Component {
 
   handleKeyDown(event) {
     const { helperVisible, options, selection } = this.state;
-    const { onKeyDown, passThroughEnter } = this.props;
+    const { onKeyDown, passThroughEnter, maxOptions } = this.props;
+    const optionsCount = maxOptions === 0 || maxOptions > options.length ? options.length : maxOptions
```
Comment (yury-dymov) [inline src/AutoCompleteTextField.js:301]: nit: `;` in the end for consistency
Also a bit cleaner to my taste would be 
`const optionsCount = maxOptions > 0 ? Math.min(options.length, maxOptions) : options.length;`
It also covers case whenever folks passing `maxOptions = -1` expecting it to be infinity.

File: src/AutoCompleteTextField.js [hunk 2]
```diff
@@ -307,11 +308,11 @@ class AutocompleteTextField extends React.Component {
           break;
         case KEY_UP:
           event.preventDefault();
-          this.setState({ selection: ((options.length + selection) - 1) % options.length });
+          this.setState({ selection: ((optionsCount + selection) - 1) % optionsCount });
```
Comment (yury-dymov) [inline src/AutoCompleteTextField.js:311]: I know it's __currently__ an unreachable state for `optionsCount` to be zero as `helperVisible` would be false but it still makes me a bit nervous.  While it's not directly related to your change can we still update logic here and below to
```
case KEY_UP:
          event.preventDefault();
          if (optionsCount > 0) {
              this.setState({ selection: Math.max(0, optionsCount + selection - 1) % optionsCount });
         }
         break;
...
          }
```

**Review by yury-dymov — CHANGES_REQUESTED** [review]
Thank you for submitting patches! Please kindly address my feedback and I will merge it shortly.

**Comment by choisee** [comment]
@yury-dymov Thanks for sincere review! Learned think about more cases. 
I applied feedbacks. Hope it could be helpful :)

**Pull request merged**